src/plugins/data/server/search/search_service.ts

```diff
@@ -41,15 +42,18 @@ interface StrategyMap {
 export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
   private searchStrategies: StrategyMap = {};
 
-  constructor(private initializerContext: PluginInitializerContext) {}
+  constructor(
+    private initializerContext: PluginInitializerContext,
+    private readonly logger: Logger
```

Comment:
not sure that we need to keep it as a separate field. `initializerContext` already has `logger` field

![image](https://user-images.githubusercontent.com/20072247/88027538-13473280-cb40-11ea-8d19-2ec4eaf89b0c.png)

Reply:
@alexwizp the one on the context is the `LoggerFactory`.
You need to initialize it with the plugin name, and it makes sense IMO to initialize it once.